activejob/lib/active_job/test_helper.rb

```diff
@@ -405,7 +407,9 @@ def assert_enqueued_with(job: nil, args: nil, at: nil, queue: nil)
         end
       end
 
-      assert matching_job, "No enqueued job found with #{expected}"
+      message = "No enqueued job found with #{expected}"
+      message += "\nPotential matches: #{potential_matches.join("\n")}" if potential_matches.present?
```

Comment:
I think `<<` is cheaper right? I would also prefer `if potential_matches.any?` here.

Reply:
note since the `# frozen_string_literal: true` comment is enabled, `<<` will fail, in which case you can do: 

message = +"No enqueued job found with #{expected}"

And then `<<` should work correctly.